help-beta/vite-plugins/pre_process_mdx_include.ts

```diff
@@ -0,0 +1,81 @@
+import fs from "node:fs";
+import path from "node:path";
+
+import type {Plugin} from "vite";
+
+const INCLUDE_REGEX = /^ {0,3}-!([^!]+)!- *$/gm;
+
+function resolveIncludes(text: string, currentDirectory: string): string {
+    return text.replaceAll(INCLUDE_REGEX, (_match, includeFileName: string) => {
+        let fullPath = path.resolve(currentDirectory, "include/" + includeFileName);
+        if (currentDirectory.endsWith("include")) {
+            // We are already inside the include directory, we don't
+            // need to append `include` in this case.
+            fullPath = path.resolve(currentDirectory, includeFileName);
+        }
+
+        if (!fs.existsSync(fullPath)) {
+            throw new Error(`Included file not found: ${fullPath}`);
+        }
+
+        const includedContent = fs.readFileSync(fullPath, "utf8");
+        return resolveIncludes(includedContent, path.dirname(fullPath));
+    });
+}
```

Comment:
I have not added circular dependency checks since we don't check for that in our original python plugin. We can add it later on if we find a lot of people getting stuck because of circular dependencies.